connexta/multi-int-store
# Enhanced logging

This PR enhances logging for HTTP requests and responses. There are many examples of how to log HTTP requests and responses in Spring boot apps, but most implementations had limitations. I did not have the time to fully investigate the many options. I consider this PR a proof-of-concept to see what  logging approaches are helpful, and what limitations are in most need of remedy.

There are several changes, all to the `multi-int-store` application. If the changes are successful there, we can look at similar changes for the other apps in the repository.

**Changes**:
~~1. Default logging level is set to ERROR. This will cut down on the noise in the logs. ~~
~~2. Logging level for com.connexta packages set to DEBUG. This is to ensure we see relevant ~~information during the active development phase of the project.~~
3.  CommonsRequestLoggingFilter to logger added to log incoming HTTP requests, ~~including the body, up to five kilobytes~~. Does not log the body.
4. Added Spring Actuator and enabled all its endpoints. Actuator traces the last 100 HTTP messages in a JSON format. To view the trace, hit the URL that ends in `/actuator/httptrace`, e.g. `http://mis:8080/actuator/httptrace`

To test common log changes:
1. Deploy the code to the cloud. 
2. Tail the MIS logs
3. Send an HTTP request to MIS.
Expected result is to see the request in the MIS log.

To test actuator changes:
1. Assuming the changes are already deployed, visit the "actuator/httptrace" endpoint.
Expected result is to see HTTP requests AND responses. NOTE: Newest messages are added to the TOP of the trace, not the bottom.
